src/Atmos/Model/AtmosModel.jl

```diff
@@ -401,11 +402,7 @@ equations.
 
     # pressure terms
     p = pressure(m, m.moisture, state, aux)
-    if m.ref_state isa HydrostaticState
-        flux.ρu += (p - aux.ref_state.p) * I
-    else
-        flux.ρu += p * I
-    end
+    flux.ρu += p * I
```

Comment:
Why is the factoring out of the hydrostatic balance in the momentum equation removed? We had subtracted this for numerical stability. Maybe I'm missing it, but I do not see where this subtraction occurs now.

Reply:
Unless I am missing something if the reference state is discretely balanced then the terms that we used to factor out should cancel out to machine precision.

Reply:
No, they only cancel out in your initial state, where ρ and p are hydrostatically balanced. But this is a special initial state, and not one that is generally maintained. In general, there is a nonhydrostatic residual, and this is essential (e.g., it's what's responsible for convection).

Reply:
So Tapio, evidence suggests that even if the reference field is discretely hydrostatically balanced using the perturbation here (and below) is more stable? Because I agree with Maciek that that with exact math, if the reference state is discretely balance this is a no-op.

Reply:
I reverted this change.